### PR TITLE
Add CRL as part of server https option

### PR DESCRIPTION
**Pull request by ericlu88**
This PR adds CRL as part of server https option. It addresses #108 partially because it does not deal with auto-bouncing the server when CRL changes.

### CODE REVIEW

**Comment by ericvw** [comment]
lgtm

**Comment by ericlu88** [comment]
Updated the PR with auto-bouncing the server when CRL is updated.

File: lib/config.ts [hunk 1]
```diff
@@ -274,9 +283,20 @@ export function get(name: string): any
...
         };
+
+        // Setup file watch for crl changes
+        fs.watch(path.resolve(__dirname, convictConf.get('https.crl')),
+                 (event: string, filename: string): void => {
+                    // Re-read the crl file
+                    otherConf['serverOptions'].httpsServerOptions.crl =
+                        fs.readFileSync(path.resolve(__dirname, convictConf.get('https.crl')));
+                    emitter.emit('change', 'https.crl');  // Signal the server that config changes
+                });
```
Comment (thmiceli) [inline lib/config.ts]: Does 'path.resolve' need to be called twice (as an argument to 'watch' and in the callback)? Can't the 'filename' argument to the call back be used instead?

Comment (ericlu88) [inline lib/config.ts]: According to [Node document:](http://nodejs.org/api/fs.html#fs_fs_watch_filename_options_listener)

```
Filename Argument#

Providing filename argument in the callback is not supported on every platform (currently it's only supported on Linux and Windows). Even on supported platforms filename is not always guaranteed to be provided. Therefore, don't assume that filename argument is always provided in the callback, and have some fallback logic if it is null.
```

But I guess I can at least cache the value in a variable so we don't need to read it multiple time.

**Comment by yupinghu** [comment]
LGTM

**Comment by ericvw** [comment]
lgtm - please squash.

**Pull request merged**